docker/tensorflow-application/video-serving/streaming.py

```diff
@@ -0,0 +1,56 @@
+from flask import Flask, Response, request
+from flask import render_template, send_file
+import argparse
+import datetime
+import imutils
+import time
+import cv2
+import requests
+import numpy as np
+from PIL import Image
+from flask import jsonify
+import time
+from collections import deque
+
+app = Flask(__name__, template_folder='templates')
+
+#Queue implemented for consistent FPS
+q=deque(maxlen=5)
+
+@app.route("/")
+def index():
+	return render_template('index.html')
+
+# TODO Use a Kafka queue which can help us to decouple the two applications 
+# (analysis and streaming) and provide a more scalable approach.
+def stream_to_queue():
+	list = request.json['instances']
+	for l in list:
+		a = np.asarray(l, dtype=np.uint32)
+		b = Image.fromarray(a.astype('uint8')) 
+		c = np.asarray(b)
+		(flag, image) = cv2.imencode(".JPEG", c)
+		q.append(image)
+
+@app.route("/video")
+def video():
+	def produce_frame():
+		while True:
+			if len(q) is 0:
+				continue
+			
+			outputFrame = q.popleft()
```

Comment:
Using a queue is a much better solution than a single global var with locks: Nice change. Not only does it allow a buffer of images to be built up, but because we are also treating each frame variable as immutable and just moving them through the data structure we'll never have any deadlocks.